### DISCUSSION TITLE
Grouping different models in the Model tree

### DISCUSSION THREAD

**Discussion by kardoka** [iTwin/itwinjs-core#5354] [Q&A] [answered]
Greetings,

Is it possible to group the different models that are loaded in the iTwin viewer? I would like to group the photogrammetry tiles seen in this picture:

![image](https://user-images.githubusercontent.com/28572657/230085657-16fa0a36-dcb7-4041-b8ce-5b49cdcb91f9.png)

Changing the [enableElementsClassGrouping](https://www.itwinjs.org/reference/appui-react/imodelcomponents/modelstreeprops/enableelementsclassgrouping/) property in [ModelsTreeProps](https://www.itwinjs.org/reference/appui-react/imodelcomponents/modelstreeprops/) does not seem to achieve this. Do I have to create a custom tree to achieve this?

**Comment by grigasp** [comment]
Hi,

The Models Tree creates a hierarchy based on information in the iModel. In your picture it seems there are 4 `Tile_*` [Subjects](https://www.itwinjs.org/bis/guide/data-organization/information-hierarchy/#the-subject-hierarchy-organizes-models), which get loaded into the hierarchy.

From your question it's not clear - is your expectation to create a different hierarchy based on the same data in the iModel, or is it to change the way data is placed in the iModel (thus affecting the way it appears in the hierarchy)?

**Reply by kardoka** [reply]
Thank you for the reply. I am open to both approaches, as long as the end result would be that the Tile subjects are grouped, even if the data in the iModel is changed.

**Reply by grigasp** [reply]
I might be subjective, but I believe it's always better to improve the data, if possible, rather than try to improve display of bad data. @abeesh, could you advice regarding grouping those `Tile_*` Subjects under a single Subject?

**Reply by kardoka** [reply]
> I might be subjective, but I believe it's always better to improve the data, if possible, rather than try to improve display of bad data. @abeesh, could you advice regarding grouping those `Tile_*` Subjects under a single Subject?

We could change the data, that would not be a problem. :)

**Reply by kardoka** [reply]
@grigasp @abeesh I have searched the iTwin.js documentation regarding Subjects, but I have not seemed to find anything related to their grouping. Could you please provide some advice regarding this?

**Reply by grigasp** [reply]
@diegoalexdiaz, maybe you can advice?

**Reply by diegoalexdiaz** [reply]
There are multiple concepts in BIS that organizes data into hierarchies. See [Information Hierarchy](https://www.itwinjs.org/bis/guide/data-organization/information-hierarchy/).

One of them is the Subject-hierarchy. It leads, at its leaves, to the containers of elements (i.e. models). That's the hierarchy that is shown in iTwin Viewer as the Model-tree.

Currently, each iModel Connector projects the external data into that hierarchy depending on how the source format is organized. @kardoka, what iModel Connector are you using? Or, are you writing/using your own Connector?

**Reply by kardoka** [reply]
Thank you @diegoalexdiaz for the reply!

For the iModel mentioned in this discussion, I am using two official connectors: IFC and MicroStation. I have created only one connection for the iModel in iTwin Synchronizer:

![image](https://user-images.githubusercontent.com/28572657/233448658-cd94f510-7143-42d5-9e39-329339623f0a.png)

I have tried to create separate connections (one for the IFC model, the other for photogrammetry), but that did not result in the photogrammetry being grouped.

**Reply by diegoalexdiaz** [reply, marked as answer]
You need to be aware of the subject-hierarchy that the current implementations of those connectors output by default in order to achieve your goal.

The IFC Connector creates one Subject per IFC file processed in a run - all at the same level (siblings).

The MicroStation Connector creates one Subject for each DgnModel processed in a run, starting from the DGN file & model specified as the root of the synchronization, aiming to reproduce the model-hierarchy that exists in the DGN dataset.

That said, if I'm understanding your need correctly (i.e. you want to see a "Photogrammetry" node above the 4 "Tile_X" subjects), I think you need to have a separate Photogrametry.dgn file that you can use as the root of the synchronization job, and reference the other 4 dgns from it.

We value that you raised your particular need to us. We will use your use-case as input into how we can improve the process in the future.

Hope it helps!
